src/snowflake/snowpark/session.py

```diff
@@ -951,6 +979,13 @@ def _resolve_packages(
                             "by ORGADMIN to use Anaconda 3rd party packages. Please follow the instructions at "
                             "https://docs.snowflake.com/en/developer-guide/udf/python/udf-python-packages.html#using-third-party-packages-from-anaconda."
                         )
+                    if not self.custom_packages_upload_enabled:
+                        raise RuntimeError(
+                            f"Cannot add package {package_req} because it is not available in Snowflake "
+                            f"and Session parameter 'custom_packages_upload_enabled' is set to False. To upload these packages, you can "
+                            f"set it to True or find the directory of these packages and add it via session.add_import(). See details at "
```

Comment:
Should True and the method name be formatted for monospace? Also, I'd use the following for the method name

> ``session.add_import`` method.